numpyro/distributions/continuous.py

```diff
@@ -491,6 +492,42 @@ def support(self):
         return constraints.greater_than(self.scale)
 
 
+class GaussianRandomWalk(Distribution):
+    arg_constraints = {'num_steps': constraints.positive, 'scale': constraints.positive}
+    support = constraints.real
+    # FIXME: cannot take grad through random.normal with dynamic shape
```

Comment:
Opps, I forgot to ask about this. Could you pls elaborate the problem so I can take a look?

Reply:
I guess you mean `grad` w.r.t. num_steps?

Reply:
That's correct. The reason why this isn't reparametrizable w.r.t. scale is because cumsum and (random.normal)'s shape depends on `num_steps` and we aren't able to take gradient through that. While I have marked it `FIXME`, this is only important if we want to use this for SVI.